src/utilities/logger.ts

```diff
@@ -0,0 +1,38 @@
+/* eslint-disable no-tabs */
+// import winston = require('winston');
+import winston from 'winston';
+import 'setimmediate';
+
+const logger = winston.createLogger({
+	level: 'info',
+	format: winston.format.simple(),
+	defaultMeta: {
+		get location() { return window.location.href; },
+		userAgent: window.navigator.userAgent,
+		service: 'webwork3'
+	},
+	transports: [
+		new winston.transports.Http({
+			level: 'error',
+			host: 'localhost',
+			port: 8080,
+			path: '/webwork3/api/utility/client-logs',
+			handleExceptions: true
+		})
+	]
+});
+
+//
+// If we're not in production then log to the console
+//
```

Comment:
I despise this style of commenting, or any time comments have pre and post comment lines.  Syntax highlighting does a great job of offsetting comments.  There is no need for anything more.